python/doc/examples/calibration/bayesian_calibration/plot_rwmh_python_distribution.py

```diff
@@ -0,0 +1,178 @@
+#
+# Then, the posterior distribution of :math:`(\alpha, \beta)`, representing the update of the prior information on :math:`(\alpha, \beta)` given the dataset has a PDF which is proportional to:
+#
+#
+# .. math::
+#   \pi(\alpha, \beta | (t_1, f_1), \ldots, (t_n, f_n) ) \propto \pi(\alpha)\pi(\beta) \left(\frac{\alpha}{\beta}\right)^{\sum_i f_i} \left(\prod_{f_i = 1} \frac{t_i}{\beta}\right)^{\alpha-1} \exp\left[-\sum_{i=1}^n\left(\frac{t_i}{\beta}\right)^\alpha\right].
+#
+# Set up the :class:`PythonDistribution`
+# --------------------------------------
+#
+# The :class:`~openturns.RandomWalkMetropolisHastings` class can be used to sample from the posterior distribution. It relies on the following objects:
+#
+# - The conditional density :math:`p(t_{1:n}|f_{1:n}, \alpha, \beta)` must be defined as a probability distribution
+# - The prior probability density :math:`\pi(\vect{\theta})`, reflecting beliefs about the possible values
+#   of :math:`\vect{\theta} = (\alpha, \beta)` before the experimental data are considered.
+# - Initial values :math:`\vect{\theta}_0` for the calibration parameters
+# - Proposal distributions used to update each parameter sequentially.
+#
+# The censured Weibuill likelihood is outside the usual catalog of probability distributions in OpenTURNS, hence we need to define it using the :class:`~openturns.PythonDistribution` class.
+
+
+# %%
+
+import numpy as np
+import openturns as ot
+from openturns.viewer import View
+ot.Log.Show(ot.Log.NONE)
+
+# %%
+# The following methods must be defined:
+#
+# - `getRange`: required for conversion to the :class:`~openturns.Distribution` format
+# - `computeLogPDF`: used by :class:`~openturns.RandomWalkMetropolisHastings` to evaluate the posterior density
+# - `setParameter` used by :class:`openturns.RandomWalkMetropolisHastings` to test new parameter values
+#
+# *Note:* We formally define a bivariate distribution on the :math:`(t_i, f_i)` couple, even though :math:`f_i` has no distibution (it is simply a covariate).
+# This is not an issue, since the sole purpose of this :class:`~openturns.PythonDistribution` object is to pass the likelihood calculation over to :class:`~openturns.RandomWalkMetropolisHastings`.
+
+# %%
+
+class CensoredWeibull(ot.PythonDistribution):
+    """
+    Right-censored Weibull log-PDF calucation
+    Each data point x is assumed 2D, with:
+        x[0]: observed functioning time
+        x[1]: nature of x[0]: 
+            if x[1]=0: x[0] is a time-to failure
+            if x[1]=1: x[0] is a censoring time
+    """
+    def __init__(self, alpha=2.0, beta=5000.0):
+        super(CensoredWeibull, self).__init__(2)
+        self.alpha = alpha
+        self.beta = beta
+    
+    def getRange(self):
+        return ot.Interval([0, 0], [1, 1], [True]*2, [False, True])
+
+    def computeLogPDF(self, x):
```

Comment:
```suggestion
    def computeLogPDF(self, x):
        if not (self.alpha>0.0 and self.beta>0.0):
            return -np.inf
```
Pour éviter les warnings en cas de proposition de alpha ou beta illicite (et aussi rendre la méthode robuste aux NaN).